src/cpp/fastdds/subscriber/SubscriberImpl.cpp

```diff
@@ -473,11 +476,23 @@ void SubscriberImpl::SubscriberReaderListener::on_liveliness_changed(
 {
     (void)status;
 
-    if (subscriber_->listener_ != nullptr)
+    SubscriberListener* listener = subscriber_->listener_;
+    if (!subscriber_->user_subscriber_->get_status_mask().is_active(StatusMask::liveliness_changed()))
```

Comment:
Should try the participant also if the listener is not set, even when the mask is active. 
```suggestion
    if (listener == nullptr || !subscriber_->user_subscriber_->get_status_mask().is_active(StatusMask::liveliness_changed()))
```